microsoft-edge/progressive-web-apps-chromium/how-to/index.md

```diff
@@ -0,0 +1,319 @@
+
+
+<!-- ====================================================================== -->
+## Step 1 - Use HTTPS
+
+Key parts of the PWA platform, such as [Service Workers][MDNServiceWorkerApi], require the use of HTTPS.  When your PWA goes live, you must publish it to an HTTPS URL.  Many hosts now offer HTTPS by default, but if your host doesn't, Let's Encrypt offers a free alternative for creating the necessary certificates.
+
+For debugging purposes, Microsoft Edge also permits `http://localhost` to use the PWA APIs.  In this tutorial, you use `http://localhost` to build your PWA.
+
+[Publish your web app as a live site][VisualStudioNodejsTutorialPublishAzureAppService], but make sure your server is configured for HTTPS.  For example, you can create an [Azure free account][AzureCreateFreeAccount].  Host your site on the [Microsoft Azure App Service][AzureWebApps] and it is served over HTTPS by default.  
+
+
+<!-- ====================================================================== -->
+## Step 2 - Create a Web App Manifest
+
+A [Web App Manifest][MDNWebAppManifest] is a JSON file containing metadata about your app, such as name, description, icons, and more.
+
+To add an app manifest to the web app:
+
+1.  In Visual Studio Code, select **File** > **Open Folder** and then select the `MySamplePwa` directory that you created earlier.
+1.  Select `Ctrl`+`N` to create a new file, and paste in the following code snippet.
+
+    ```json
+    {
+        "lang": "en-us",
+        "name": "My Sample PWA",
+        "short_name": "SamplePWA",
+        "description": "A sample PWA for testing purposes",
+        "start_url": "/",
+        "background_color": "#2f3d58",
+        "theme_color": "#2f3d58",
+        "orientation": "any",
+        "display": "standalone",
+        "icons": [
+            {
+                "src": "/icon512.png",
+                "sizes": "512x512"
+            }
+        ]
+    }
+    ```
+
+1.  Save the file as `/MySamplePwa/public/manifest.json`.
+1.  Add a 512x512 app icon image named `icon512.png` to `/MySamplePwa/public/images`.  You can use the [sample image](../media/progressive-web-app.png) for testing purposes.
+1.  In Visual Studio Code, open `/public/index.html`, and add the following code snippet inside the `<head>` tag.
+    
+    ```html
+    <link rel="manifest" href="/manifest.json">
+    ```
+
+
+<!-- ====================================================================== -->
+## Step 3 - Add a Service Worker
+
+Service workers are the key technology behind PWAs, enabling scenarios that were previously limited to native apps, such as offline support, advanced caching, and running background tasks.
+
+Service workers are background tasks that intercept network requests from your web app.  Service workers attempt to complete tasks, even when your PWA is not running.  Tasks include:
```

Comment:
Service workers are a specialized web worker and, as such, they run in a separate thread from the main app. They can execute background tasks, but they are not, themselves, a task.